### PR TITLE
Made a generic profile avatar widget, reused in all places that need …

### PR DESCRIPTION
**Pull request by LeeMatthewHiggins**
[FARM-318 - Profile avatar updated for reuse across app]

#### 📲 What

Made sure the avatar widget is the same implementation all over the app

#### 🤔 Why
		
Its used in multiple spots and needs to change dynamically on profile change. 
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/model/bloc/profile/ProfileListItemViewModelTransformer.dart [hunk 1]
```diff
@@ -10,10 +14,13 @@ class SwitchProfileListItemViewModelTransformer
   SwitchProfileListItemViewModelTransformer(this._switch, this._currentProfile);
   @override
   SwitchProfileListItemViewModel transform({ProfileEntity from}) {
+    final profileViewModel = ProfileViewModel(loadingStatus: LoadingStatus.SUCCESS,initials: PersonName(from.name).initials,image: from.avatar);
```
Comment (alexanishchenko88) [inline lib/model/bloc/profile/ProfileListItemViewModelTransformer.dart:17]: Try to format this adding coma just to the end of inputs. This makes code more readable without need of horizontal scroll 😄

File: lib/ui/common/ProfileAvatar.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import 'image_provider_view.dart';
+
+class ProfileAvatar extends StatelessWidget {
+  final ViewModelProvider<ProfileViewModel> _viewModelProvider;
+  final double _width;
+  final double _height;
+  final Color _backgroundColor;
+  final TextStyle _textStyle;
+
+  const ProfileAvatar({Key key,ViewModelProvider<ProfileViewModel> viewModelProvider, double width, double height, TextStyle textStyle, Color backgroundColor}) : this._viewModelProvider = viewModelProvider, this._width = width, this._height=height, this._textStyle= textStyle,this._backgroundColor = backgroundColor, super(key: key);
```
Comment (alexanishchenko88) [inline lib/ui/common/ProfileAvatar.dart]: As above, this is very difficult to read, just apply the format 👍

File: lib/ui/common/ProfileAvatar.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+              height: _height,
+            );
+          }
+          return Container(
+              height: _width,
+              width: _height,
+              child: ClipOval(
+                child: Container(
+                  child: avatarWidget,
+                  color: _backgroundColor ?? Color(0xffe9eaf2),
```
Comment (alexanishchenko88) [inline lib/ui/common/ProfileAvatar.dart]: Maybe move this color to const?

File: lib/ui/home.dart [hunk 4]
```diff
@@ -212,8 +213,20 @@ class Home extends StatelessWidget {
...
 
-  //TODO Build it properly
-  TabNavigator _buildTabNavigatorWithCircleImageWidget(Widget page) {
+  Widget _buildProfileIcon(HomeViewModel viewModel) {
+    return ProfileAvatar(
+      viewModelProvider: ProfileDetailProvider(
+          profileRepo: viewModel.currentProfile,
+          plotRepo:
+              repositoryProvider.getMyPlotRepository(viewModel.currentProfile)),
+      width: _Constants.iconHeight,
```
Comment (alexanishchenko88) [inline lib/ui/home.dart:222]: Is this correct or should be iconWidth?

File: lib/ui/home.dart [hunk 5]
```diff
@@ -223,16 +236,14 @@ class Home extends StatelessWidget {
             shape: BoxShape.circle,
           ),
           padding: EdgeInsets.all(2.0),
-          height: 27,
-          child: CircleAvatar(
-            child: Image.asset('assets/raw/mock_profile_image.png'),
-          ),
+          height: _Constants.iconHeight,
+          width: _Constants.iconHeight,
```
Comment (alexanishchenko88) [inline lib/ui/home.dart:240]: The same here

File: lib/ui/home.dart [hunk 6]
```diff
@@ -223,16 +236,14 @@ class Home extends StatelessWidget {
...
+          child: profileIcon,
         ),
         icon: Container(
-          height: 27,
-          child: CircleAvatar(
-            child: Image.asset('assets/raw/mock_profile_image.png'),
-          ),
+          height: _Constants.iconHeight,
+          width: _Constants.iconHeight,
```
Comment (alexanishchenko88) [inline lib/ui/home.dart:245]: And here as well

**Review by alexanishchenko88 — COMMENTED** [review]
Apparently this code is not well formatted, with android studio we have a keyboard shortcut what do it for us or you can set in preferences indicating format applying after save. 
Basically that gives as a more readable code.

File: lib/ui/home.dart (hunk 4 above)
Comment (albertmiro) [reply lib/ui/home.dart:222]: Probably this builds a circle avatar with same width and height for the image, so the width and height would be the same size. Maybe I would rename it to `iconSize`?

File: lib/ui/home.dart (hunk 5 above)
Comment (albertmiro) [reply lib/ui/home.dart:240]: I think is the same reason as before.

File: lib/ui/home.dart (hunk 6 above)
Comment (albertmiro) [reply lib/ui/home.dart:245]: The same ;-)

**Review by albertmiro — DISMISSED** [review]

File: lib/ui/home.dart (hunk 4 above)
Comment (LeeMatthewHiggins) [reply lib/ui/home.dart:222]: just used existing code. I´ll rename tho

File: lib/model/bloc/profile/ProfileListItemViewModelTransformer.dart (hunk 1 above)
Comment (LeeMatthewHiggins) [reply lib/model/bloc/profile/ProfileListItemViewModelTransformer.dart:17]: missing the comma, which stops the auto format working.

**Review by albertmiro — APPROVED** [review]

**Review by alexanishchenko88 — APPROVED** [review]

File: lib/ui/common/ProfileAvatar.dart (hunk 2 above)
Comment (LeeMatthewHiggins) [reply lib/ui/common/ProfileAvatar.dart]: sorted 👍

File: lib/ui/common/ProfileAvatar.dart (hunk 3 above)
Comment (LeeMatthewHiggins) [reply lib/ui/common/ProfileAvatar.dart]: done 👍

File: lib/ui/home.dart (hunk 4 above)
Comment (LeeMatthewHiggins) [reply lib/ui/home.dart:222]: done 👍

File: lib/ui/home.dart (hunk 5 above)
Comment (LeeMatthewHiggins) [reply lib/ui/home.dart:240]: sorted 👍

File: lib/ui/home.dart (hunk 6 above)
Comment (LeeMatthewHiggins) [reply lib/ui/home.dart:245]: sorted 👍

File: lib/model/bloc/profile/ProfileListItemViewModelTransformer.dart (hunk 1 above)
Comment (LeeMatthewHiggins) [reply lib/model/bloc/profile/ProfileListItemViewModelTransformer.dart:17]: sorted 👍

**Pull request merged**